tests/test_utils.py

```diff
@@ -233,6 +256,136 @@ def test_user_with_multiple_contexts_has_no_access_via_jwt(self):
             'some_totally_different_context'
         )
 
+    @ddt.data(
+        (['coupon-management', 'data_api_access'], {'some_context', 'some_other_context'}),
+        # The 'coupon-management' feature role is provided by both system roles granted in our toy JWT.
+        (['coupon-management'], {'some_context', 'some_other_context'}),
+        # The 'data_api_access' feature role is provided only by the 'enterprise_admin' system role
+        # granted in the toy JWT.
+        (['data_api_access'], {'some_other_context'}),
+        ([], set()),
+        (['not-a-feature-role', 'data_api_access'], {'some_other_context'}),
+        (['not-a-feature-role'], set()),
+    )
+    @ddt.unpack
+    def test_contexts_accessible_from_request(self, role_names, expected_contexts):
+        """
+        For various role names, `contexts_accessible_from_request()` should grant access to certain contexts.
+        """
+        toy_decoded_jwt = {
+            "roles": [
+                "coupon-manager:some_context",  # provides the 'coupon-management' feature role
+                "enterprise_admin:some_other_context",  # provides the 'data_api_access' feature role
+            ]
+        }
+        with mock.patch('edx_rbac.utils.get_decoded_jwt', return_value=toy_decoded_jwt):
+            request = mock.Mock()
+            assert expected_contexts == contexts_accessible_from_request(request, role_names)
+
+    @ddt.data(
+        ['coupon-management', 'data_api_access'],
+        [],
+        ['coupone-management'],
+        ['data_api_access'],
+        ['not-a-feature-role', 'data_api_access'],
+        ['not-a-feature-role'],
+    )
+    def test_contexts_accessible_from_request_with_no_jwt_roles(self, role_names):
+        """
+        No contexts should ever be accessible from a JWT if the JWT provides no roles.
+        """
+        toy_decoded_jwt = {
+            "roles": []
+        }
+        with mock.patch('edx_rbac.utils.get_decoded_jwt', return_value=toy_decoded_jwt):
+            request = mock.Mock()
+            assert set() == contexts_accessible_from_request(request, role_names)
+
+    @ddt.data(
+        (list, True),
+        (str, True),
+        (set, True),
+        (dict, True),
+        (tuple, True),
+        (int, False),
+        (float, False),
+        (bool, False),
+        (bytes, True),
+        (lambda: None, False),
+    )
+    @ddt.unpack
+    def test_is_iterable(self, func, expected_value):
+        assert expected_value == is_iterable(func())
+
+    @ddt.data(
+        (None, {None}),
+        ('abcabc', {'abcabc'}),
+        ('', {''}),
+        ('a', {'a'}),
+        (['a', 'b', 'b'], {'a', 'b'}),
+        ({'a', 'b', 'd'}, {'a', 'b', 'd'}),
+        ({'a': 1, 'b': 2, 'c': 3}, {'a', 'b', 'c'}),
+        (('a', 'b', 'c', 'c'), {'a', 'b', 'c'}),
+        ([None], {None}),
+        (((1, 2), (3, 4)), {(1, 2), (3, 4)}),  # tuples are hashable, and can be elements of a set
+    )
```

Comment:
Not at all on you, but I always wish there was a nicer way to write ddt. I always find it so hard to read, but oh well

Reply:
good call, you can actually unpack dicts:
```
@unpack
@data({'first': 1, 'second': 3, 'third': 2},
          {'first': 4, 'second': 6, 'third': 5})
```
I'll do that here and for some of the other more onerous ones.